service/matching/matching_engine.go

```diff
@@ -676,6 +674,21 @@ pollLoop:
 					tag.BuildId(requestClone.WorkerVersionCapabilities.GetBuildId()),
 				)
 				task.finish(nil, false)
+			case *serviceerrors.ObsoleteMatchingTask:
+				// History should've scheduled another task on the right task queue and deployment.
+				// Dropping this one.
+				e.logger.Info("dropping obsolete workflow task",
+					tag.WorkflowTaskQueueName(taskQueueName),
```

Comment:
the log tag boilerplate in this function is getting a little out of hand. let's add some helpers later?